speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java

```diff
@@ -0,0 +1,213 @@
+import javax.sound.sampled.DataLine;
+import javax.sound.sampled.TargetDataLine;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.List;
+
+/**
+ * Google Cloud SpeechToText API sample application. Example usage: mvn package exec:java
+ * -Dexec.mainClass='com.example.speech.MicStreamRecognize' -Dexec.args="micstreamrecognize
+ * <duration>"
+ */
+public class MicStreamRecognize {
+
+  /**
+   * Demonstrates using the Speech to Text client to convert Microphone streaming speech to text.
+   *
+   * @throws Exception on SpeechToTextClient Errors.
+   */
+  // Microphone audio format specification
+  private static AudioFormat format = new AudioFormat(16000, 16, 1, true, false);
+
+  private static DataLine.Info targetInfo = new DataLine.Info(TargetDataLine.class, format);
+  private static TargetDataLine targetLine;
+  private static int BYTES_PER_BUFFER = 6400; // buffer size in bytes
+  // Creating shared object
+  private static volatile BlockingQueue<byte[]> sharedQueue = new LinkedBlockingQueue();
+
+  public static void main(String... args) throws Exception {
+    if (args.length < 1) {
+      System.out.println("Usage:");
+      System.out.printf(
+          "\tjava %s \"<command>\" \"<duration>\"\n"
+              + "Command:\n"
+              + "\tmicstreamrecognize\n"
+              + "Duration(optional):\n\tIn seconds.(Maximum of 60 seconds)\n",
+          MicStreamRecognize.class.getCanonicalName());
+      return;
+    }
+    String command = args[0];
+    Integer duration = args.length > 1 ? Integer.parseInt(args[1]) : 10;
+
+    // Use command to invoke transcription
+    if (command.equals("micstreamrecognize")) {
+      micRecognize(duration);
+    }
+  }
+
+  /**
+   * Performs streaming speech recognition on microphone audio data.
+   *
+   * @param duration the time duration for the microphone streaming
+   */
+  public static void micRecognize(Integer duration) throws Exception {
+    // Creating microphone input buffer thread
+    micBuffer micrunnable = new micBuffer();
+    Thread micThread = new Thread(micrunnable);
+    int durationMillSec = duration * 1000;
+    if (!AudioSystem.isLineSupported(targetInfo)) {
+      System.out.println("Microphone not supported");
+      System.exit(0);
+    }
+    // Instantiates a client with GOOGLE_APPLICATION_CREDENTIALS
+    try (SpeechClient speech = SpeechClient.create()) {
+
+      // Configure request with local raw PCM audio
+      RecognitionConfig recConfig =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(16000)
+              .setModel("default")
+              .build();
+      StreamingRecognitionConfig config =
+          StreamingRecognitionConfig.newBuilder().setConfig(recConfig).build();
+
+      class ResponseApiStreamingObserver<T> implements ApiStreamObserver<T> {
```

Comment:
Unless I'm missing something, I don't see any need for this class to be defined during the execution of this method. Please move the definition outside of the method body (or provide comments explaining why it is defined here, if necessary).